packages/expo-av/ios/EXAV/EXAV.m

```diff
@@ -215,7 +215,9 @@ +(NSString *)moduleName
 // we just need this to dismiss that warning.
 + (BOOL)requiresMainQueueSetup
 {
-  return NO;
+  // We are now using main thread to avoid thread safety issues with `EXAVPlayerData` and `EXVideoView`
+  // return `YES` to avoid deadlock warnings.
+  return YES;
```

Comment:
I doubt that this implementation is used by our expo/unimodule architecture too.
See the comment two lines above, it was only set to make `RCTBridgeModule` happy and I suspect this might be dead code since this isn't a plain RN module.
So I'm afraid this won't give the effect you're trying to achieve and makes any difference. But the good news is that implementing `methodQueue` (which you did below) should be sufficient 😁

Reply:
The `methodQueue` is used here https://github.com/expo/expo/blob/fbeba07e6f6388a611b52c99b91450d9121514ee/packages/expo-modules-core/ios/NativeModulesProxy/EXNativeModulesProxy.mm#L194 😉 
In the same way the bridge modules do.

Reply:
That's right, and looking at the `EXNativeModulesProxy`, the `requiresMainQueueSetup` in `EXAV` is a no-op 😁